### DISCUSSION TITLE
Hands swapped with Varjo XR-4

### DISCUSSION THREAD

**Discussion by paulmelis** [StereoKit/StereoKit#1031] [Q&A]
Before I bug Varjo Support to see if this is an issue in their OpenXR layer, just wanted to ask if swapped hands is something that sounds familiar? I'm on 0.3.9 and other non-StereoKit apps seem to work fine when it comes to hand tracking. I've successfully used SK on the XR-3, but the software for the XR-4 has been quite a mess, with every new release fixing important stuff (like decent tracking), but then breaking something else. This is with the separate UltraLeap hand tracking module, btw, as the XR-4 inexplicitly no longer comes with hand tracking built in.

Edit: when picking up the controllers the hands do move to the correct left/right position and orientation, it's only hand tracked interaction that's wrong

**Comment by paulmelis** [comment]
Well, after trying some different SK apps this seems to have spontaneously fixed itself... weird

**Reply by maluoi** [reply]
Getting hands confused, or flipping them backwards does seem to be a fairly common sort of issue in hand tracking systems! I guess if you're getting this a lot, it might be worth evaluating the lighting situation in your environment. For IR based systems in particular, I know I've had contaminating light sources that I didn't notice for way too long :D

**Reply by paulmelis** [reply]
Well, just as easy as the problem solved itself it's now suddenly back in all SK apps. If I try one of the Varjo demos right after then hand orientations and overlays are correct (and those demos use the Unity layer, not OpenXR). So it seems to be something in the Varjo OpenXR layer :-/

**Reply by paulmelis** [reply]
Ah! At least I found a Varjo demo that's apparently also based on OpenXR (the Lamborghini one). It shows the same issue, so now I have something to report to them 😄

**Reply by paulmelis** [reply]
Weird, it seems to be caused by running a non-OpenXR app (in this case a Varjo demo built with Unity) that screws up the hand tracking in OpenXR. Only after a Varjo system restart does it get fixed.
